tests/commands/test_compile.py

```diff
@@ -112,3 +112,28 @@ def test__compile_account_contract(mock_subprocess):
         stdout=mock_subprocess.PIPE,
     )
     mock_process.communicate.assert_called_once()
+
+
+def test__compile_contract_without_hint_validation(mock_subprocess):
+    contract_name_root = "contract_with_unwhitelisted_hints"
+    path = f"path/to/{contract_name_root}.cairo"
+
+    mock_process = Mock()
+    mock_subprocess.Popen.return_value = mock_process
+
+    _compile_contract(path, disable_hint_validation=True)
+
+    mock_subprocess.Popen.assert_called_once_with(
+        [
+            "starknet-compile",
+            path,
+            f"--cairo_path={CONTRACTS_DIRECTORY}",
+            "--output",
+            f"artifacts/{contract_name_root}.json",
+            "--abi",
+            f"artifacts/abis/{contract_name_root}.json",
+            "--disable_hint_validation",
+        ],
+        stdout=mock_subprocess.PIPE,
+    )
+    mock_process.communicate.assert_called_once()
```

Comment:
is this enough to confirm it ran successfully? so it makes sense to have an example contract

Reply:
I'm not used to testing with `Mock` so i don't know. I used the same test as for the account flag. To me it would make sense to assert that the command exited with 0 and to try compiling a contract with unwhitelisted hints without adding the flag and to assert that it failed but once more i'm not really familiar with the mock class